packages/python/protocol/protocol.py

```diff
@@ -310,3 +311,62 @@ class PocketNetworkMongoDBConfig(BaseModel):
 
     class Config:
         arbitrary_types_allowed = True
+
+
+######################
+# TIMEOUT HANDLER
+######################
+
+
+class TTFT(BaseModel):
+    x: List[int]
+    y: List[int]
+
+
+class LLMTimeouts(BaseModel):
+    ttft: TTFT
+    tpot: float
+    queue: float
+
+
+class TimeoutHandler(BaseModel):
+    service: str
+    timeouts: LLMTimeouts
+
+    # Define the functions for each chain
+    def timeouts_a100(self, prefill: int, decode: int) -> float:
+        x = self.timeouts.ttft.x
+        y = self.timeouts.ttft.y
+        queue = self.timeouts.queue
+        z = np.polyfit(x, y, 2)
```

Comment:
you are calling polyfit on each timeout calculation request?

Also, handle when this polyfit fails (configuration can be wrong)

Reply:
If the timeout correspond to a 'llm', now it only fit one time.